python/pyarrow/tests/test_convert_pandas.py

```diff
@@ -160,9 +160,40 @@ def test_integer_index_column(self):
         df = pd.DataFrame([(1, 'a'), (2, 'b'), (3, 'c')])
         _check_pandas_roundtrip(df, preserve_index=True)
 
+    def test_index_metadata_field_name(self):
+        # test None case, and strangely named non-index columns
+        df = pd.DataFrame(
+            [(1, 'a', 3.1), (2, 'b', 2.2), (3, 'c', 1.3)],
+            index=pd.MultiIndex.from_arrays(
+                [['c', 'b', 'a'], [3, 2, 1]],
+                names=[None, 'foo']
+            )
+        ).rename(columns=dict(zip(range(3), ['a', None, '__index_level_0__'])))
```

Comment:
not that important, but doing `columns= ['a', None, '__index_level_0__']` inside the `DataFrame` call is a bit simpler

Reply:
Yep, thank you. That is much better.